packages/test/test-end-to-end-tests/src/test/container.spec.ts

```diff
@@ -666,6 +669,92 @@ describeNoCompat("Container", (getTestObjectProvider) => {
 		assert.strictEqual(deltaManagerClosed, 1, "DeltaManager should send closed event");
 		assert.strictEqual(runtimeDispose, 1, "ContainerRuntime should send dispose event");
 	});
+
+	it("clientDetailsOverride does not cause client details of other containers with the same loader to change", async function () {
+		const documentId = uuid();
+		const client: IClient = {
+			details: {
+				capabilities: { interactive: true },
+			},
+			permission: [],
+			scopes: [],
+			user: { id: "" },
+			mode: "write",
+		};
+		const loaderProps: Partial<ILoaderProps> = {
+			options: {
+				client,
+			},
+		};
+		const loader = provider.makeTestLoader({ loaderProps });
+		const container1 = await loader.createDetachedContainer(provider.defaultCodeDetails);
+		const createNewRequest = provider.driver.createCreateNewRequest(documentId);
+		await container1.attach(createNewRequest);
+
+		// Check that client details are the expected ones before resolving a second container with different client details
+		assert.equal(
+			(container1 as any).clientDetails?.capabilities?.interactive,
+			true,
+			"Interactive container's capabilities do not say 'interactive: true' before resolving second container",
```

Comment:
Nit: for assert messages I prefer to phrase as "should", since different engineers mean "capabilities do not say 'interactive: true' (but should)" vs. "(I expect that) capabilities do not say 'interactive: true' (but they do)".  Whereas "capabilities should say 'interactive: true'" is unambiguous.